### ISSUE TITLE
Creeper going into portal at the same time as exploding causes a crash

### ISSUE THREAD

**Issue by lexi-the-cute** [kyrptonaught/customportalapi#88]
```
[08:16:00] [Server thread/ERROR]: Encountered an unexpected exception
net.minecraft.class_148: Colliding entity with block
        at net.minecraft.server.MinecraftServer.method_3813(MinecraftServer.java:901) ~[server-intermediary.jar:?]
        at net.minecraft.class_3176.method_3813(class_3176.java:283) ~[server-intermediary.jar:?]
        at net.minecraft.server.MinecraftServer.method_3748(MinecraftServer.java:824) ~[server-intermediary.jar:?]
        at net.minecraft.server.MinecraftServer.handler$zlh000$carpet$modifiedRunLoop(MinecraftServer.java:12942) ~[server-intermediary.jar:?]
        at net.minecraft.server.MinecraftServer.method_29741(MinecraftServer.java:650) ~[server-intermediary.jar:?]
        at net.minecraft.server.MinecraftServer.method_29739(MinecraftServer.java:265) ~[server-intermediary.jar:?]
        at java.lang.Thread.run(Thread.java:842) ~[?:?]
Caused by: java.lang.NullPointerException: Cannot invoke "Object.hashCode()" because "key" is null
        at java.util.concurrent.ConcurrentHashMap.get(ConcurrentHashMap.java:936) ~[?:?]
        at net.kyrptonaught.customportalapi.portal.linking.PortalLinkingStorage.getDestination(PortalLinkingStorage.java:50) ~[customportalapi-0.0.1-beta64-1.20-1536c58c09495c99.jar:?]
        at net.kyrptonaught.customportalapi.util.CustomTeleporter.customTPTarget(CustomTeleporter.java:65) ~[customportalapi-0.0.1-beta64-1.20-1536c58c09495c99.jar:?]
        at net.kyrptonaught.customportalapi.util.CustomTeleporter.TPToDim(CustomTeleporter.java:46) ~[customportalapi-0.0.1-beta64-1.20-1536c58c09495c99.jar:?]
        at net.kyrptonaught.customportalapi.CustomPortalBlock.method_9548(CustomPortalBlock.java:110) ~[customportalapi-0.0.1-beta64-1.20-1536c58c09495c99.jar:?]
        at net.minecraft.class_4970$class_4971.method_26178(class_4970.java:1121) ~[server-intermediary.jar:?]
        at net.minecraft.class_1297.redirect$cce000$carpet-fixes$checkBlockCollisionBetter(class_1297.java:9384) ~[server-intermediary.jar:?]
        at net.minecraft.class_1297.method_5852(class_1297.java:1137) ~[server-intermediary.jar:?]
        at net.minecraft.class_1297.redirect$cce000$carpet-fixes$onEntityCollision(class_1297.java:9352) ~[server-intermediary.jar:?]
        at net.minecraft.class_1297.method_5784(class_1297.java:799) ~[server-intermediary.jar:?]
        at net.minecraft.class_1542.method_5773(class_1542.java:136) ~[server-intermediary.jar:?]
        at net.minecraft.class_3218.method_18762(class_3218.java:739) ~[server-intermediary.jar:?]
        at net.minecraft.class_1937.method_18472(class_1937.java:480) ~[server-intermediary.jar:?]
        at net.minecraft.class_3218.method_31420(class_3218.java:385) ~[server-intermediary.jar:?]
        at net.minecraft.class_5574.method_31791(class_5574.java:54) ~[server-intermediary.jar:?]
        at net.minecraft.class_3218.method_18765(class_3218.java:353) ~[server-intermediary.jar:?]
```

**Comment by lexi-the-cute** [comment]
turns out just going through a busted portal causes the crash too

```
net.minecraft.class_148: Colliding entity with block
        at net.minecraft.class_3222.method_14226(class_3222.java:574) ~[server-intermediary.jar:?]
        at net.minecraft.class_3244.method_18784(class_3244.java:269) ~[server-intermediary.jar:?]
        at net.minecraft.class_2535.method_10754(class_2535.java:259) ~[server-intermediary.jar:?]
        at net.minecraft.class_3242.method_14357(class_3242.java:172) ~[server-intermediary.jar:?]
        at net.minecraft.server.MinecraftServer.method_3813(MinecraftServer.java:908) ~[server-intermediary.jar:?]
        at net.minecraft.class_3176.method_3813(class_3176.java:283) ~[server-intermediary.jar:?]
        at net.minecraft.server.MinecraftServer.method_3748(MinecraftServer.java:824) ~[server-intermediary.jar:?]
        at net.minecraft.server.MinecraftServer.handler$zlh000$carpet$modifiedRunLoop(MinecraftServer.java:12942) ~[server-intermediary.jar:?]
        at net.minecraft.server.MinecraftServer.method_29741(MinecraftServer.java:650) ~[server-intermediary.jar:?]
        at net.minecraft.server.MinecraftServer.method_29739(MinecraftServer.java:265) ~[server-intermediary.jar:?]
        at java.lang.Thread.run(Thread.java:842) ~[?:?]
Caused by: java.lang.NullPointerException: Cannot invoke "Object.hashCode()" because "key" is null
        at java.util.concurrent.ConcurrentHashMap.get(ConcurrentHashMap.java:936) ~[?:?]
        at net.kyrptonaught.customportalapi.portal.linking.PortalLinkingStorage.getDestination(PortalLinkingStorage.java:50) ~[customportalapi-0.0.1-beta64-1.20-1536c58c09495c99.jar:?]
        at net.kyrptonaught.customportalapi.util.CustomTeleporter.customTPTarget(CustomTeleporter.java:65) ~[customportalapi-0.0.1-beta64-1.20-1536c58c09495c99.jar:?]
        at net.kyrptonaught.customportalapi.util.CustomTeleporter.TPToDim(CustomTeleporter.java:46) ~[customportalapi-0.0.1-beta64-1.20-1536c58c09495c99.jar:?]
        at net.kyrptonaught.customportalapi.CustomPortalBlock.method_9548(CustomPortalBlock.java:110) ~[customportalapi-0.0.1-beta64-1.20-1536c58c09495c99.jar:?]
        at net.minecraft.class_4970$class_4971.method_26178(class_4970.java:1121) ~[server-intermediary.jar:?]
        at net.minecraft.class_1297.redirect$cce000$carpet-fixes$checkBlockCollisionBetter(class_1297.java:9384) ~[server-intermediary.jar:?]
        at net.minecraft.class_1297.method_5852(class_1297.java:1137) ~[server-intermediary.jar:?]
        at net.minecraft.class_1297.redirect$cce000$carpet-fixes$onEntityCollision(class_1297.java:9352) ~[server-intermediary.jar:?]
        at net.minecraft.class_1297.method_5784(class_1297.java:799) ~[server-intermediary.jar:?]
        at net.minecraft.class_1309.method_26318(class_1309.java:2312) ~[server-intermediary.jar:?]
        at net.minecraft.class_1309.method_6091(class_1309.java:2232) ~[server-intermediary.jar:?]
        at net.minecraft.class_1657.method_6091(class_1657.java:1614) ~[server-intermediary.jar:?]
        at net.minecraft.class_1309.method_6007(class_1309.java:2719) ~[server-intermediary.jar:?]
        at net.minecraft.class_1657.method_6007(class_1657.java:571) ~[server-intermediary.jar:?]
        at net.minecraft.class_1309.method_5773(class_1309.java:2419) ~[server-intermediary.jar:?]
        at net.minecraft.class_1657.method_5773(class_1657.java:283) ~[server-intermediary.jar:?]
        at net.minecraft.class_3222.method_14226(class_3222
.java:510) ~[server-intermediary.jar:?]
        ... 10 more
```

![2023-12-15_03 22 30](https://github.com/kyrptonaught/customportalapi/assets/120563837/7a1a0b07-fc2a-44b6-bac2-994151a56d3b)

i did save a backup of the world, but i have to make sure to remove anything irrelevant as the server is private and could contain personal information

**Comment by warior456** [comment]
i found the same bug i'll add some extra info:
option 1 : when an entity pathfinds to somewhere behind the portal and runs trough it it crashes
option 2: when an entity is in 2 portal blocks at the same time it crashes (only on half bugged portals)
extra logs: ```java
```
[17:30:46] [Server thread/ERROR] (Minecraft) Encountered an unexpected exception
 net.minecraft.util.crash.CrashException: Colliding entity with block
	at net.minecraft.server.MinecraftServer.tickWorlds(MinecraftServer.java:860) ~[minecraft-merged-project-root-1.20.1-net.fabricmc.yarn.1_20_1.1.20.1+build.1-v2.jar:?]
	at net.minecraft.server.MinecraftServer.tick(MinecraftServer.java:798) ~[minecraft-merged-project-root-1.20.1-net.fabricmc.yarn.1_20_1.1.20.1+build.1-v2.jar:?]
	at net.minecraft.server.integrated.IntegratedServer.tick(IntegratedServer.java:99) ~[minecraft-merged-project-root-1.20.1-net.fabricmc.yarn.1_20_1.1.20.1+build.1-v2.jar:?]
	at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:658) ~[minecraft-merged-project-root-1.20.1-net.fabricmc.yarn.1_20_1.1.20.1+build.1-v2.jar:?]
	at net.minecraft.server.MinecraftServer.method_29739(MinecraftServer.java:288) ~[minecraft-merged-project-root-1.20.1-net.fabricmc.yarn.1_20_1.1.20.1+build.1-v2.jar:?]
	at java.lang.Thread.run(Thread.java:833) ~[?:?]
Caused by: java.lang.NullPointerException: Cannot invoke "Object.hashCode()" because "key" is null
	at java.util.concurrent.ConcurrentHashMap.get(ConcurrentHashMap.java:936) ~[?:?]
	at net.kyrptonaught.customportalapi.portal.linking.PortalLinkingStorage.getDestination(PortalLinkingStorage.java:50) ~[customportalapi-0.0.1-beta64-1.20.jar:?]
	at net.kyrptonaught.customportalapi.util.CustomTeleporter.customTPTarget(CustomTeleporter.java:65) ~[customportalapi-0.0.1-beta64-1.20.jar:?]
	at net.kyrptonaught.customportalapi.util.CustomTeleporter.TPToDim(CustomTeleporter.java:46) ~[customportalapi-0.0.1-beta64-1.20.jar:?]
	at net.kyrptonaught.customportalapi.CustomPortalBlock.onEntityCollision(CustomPortalBlock.java:110) ~[customportalapi-0.0.1-beta64-1.20.jar:?]
	at net.minecraft.block.AbstractBlock$AbstractBlockState.onEntityCollision(AbstractBlock.java:1106) ~[minecraft-merged-project-root-1.20.1-net.fabricmc.yarn.1_20_1.1.20.1+build.1-v2.jar:?]
	at net.minecraft.entity.Entity.checkBlockCollision(Entity.java:1279) ~[minecraft-merged-project-root-1.20.1-net.fabricmc.yarn.1_20_1.1.20.1+build.1-v2.jar:?]
	at net.minecraft.entity.Entity.tryCheckBlockCollision(Entity.java:1041) ~[minecraft-merged-project-root-1.20.1-net.fabricmc.yarn.1_20_1.1.20.1+build.1-v2.jar:?]
	at net.minecraft.entity.Entity.move(Entity.java:997) ~[minecraft-merged-project-root-1.20.1-net.fabricmc.yarn.1_20_1.1.20.1+build.1-v2.jar:?]
	at net.minecraft.entity.LivingEntity.applyMovementInput(LivingEntity.java:2156) ~[minecraft-merged-project-root-1.20.1-net.fabricmc.yarn.1_20_1.1.20.1+build.1-v2.jar:?]
	at net.minecraft.entity.LivingEntity.travel(LivingEntity.java:2098) ~[minecraft-merged-project-root-1.20.1-net.fabricmc.yarn.1_20_1.1.20.1+build.1-v2.jar:?]
	at net.minecraft.entity.LivingEntity.tickMovement(LivingEntity.java:2515) ~[minecraft-merged-project-root-1.20.1-net.fabricmc.yarn.1_20_1.1.20.1+build.1-v2.jar:?]
	at net.minecraft.entity.mob.MobEntity.tickMovement(MobEntity.java:544) ~[minecraft-merged-project-root-1.20.1-net.fabricmc.yarn.1_20_1.1.20.1+build.1-v2.jar:?]
	at net.minecraft.entity.passive.PassiveEntity.tickMovement(PassiveEntity.java:128) ~[minecraft-merged-project-root-1.20.1-net.fabricmc.yarn.1_20_1.1.20.1+build.1-v2.jar:?]
	at net.minecraft.entity.passive.AnimalEntity.tickMovement(AnimalEntity.java:60) ~[minecraft-merged-project-root-1.20.1-net.fabricmc.yarn.1_20_1.1.20.1+build.1-v2.jar:?]
	at net.minecraft.entity.passive.AbstractHorseEntity.tickMovement(AbstractHorseEntity.java:551) ~[minecraft-merged-project-root-1.20.1-net.fabricmc.yarn.1_20_1.1.20.1+build.1-v2.jar:?]
	at net.minecraft.entity.LivingEntity.tick(LivingEntity.java:2249) ~[minecraft-merged-project-root-1.20.1-net.fabricmc.yarn.1_20_1.1.20.1+build.1-v2.jar:?]
	at net.minecraft.entity.mob.MobEntity.tick(MobEntity.java:368) ~[minecraft-merged-project-root-1.20.1-net.fabricmc.yarn.1_20_1.1.20.1+build.1-v2.jar:?]
	at net.minecraft.entity.passive.AbstractHorseEntity.tick(AbstractHorseEntity.java:580) ~[minecraft-merged-project-root-1.20.1-net.fabricmc.yarn.1_20_1.1.20.1+build.1-v2.jar:?]
	at net.minecraft.server.world.ServerWorld.tickEntity(ServerWorld.java:654) ~[minecraft-merged-project-root-1.20.1-net.fabricmc.yarn.1_20_1.1.20.1+build.1-v2.jar:?]
	at net.minecraft.world.World.tickEntity(World.java:481) ~[minecraft-merged-project-root-1.20.1-net.fabricmc.yarn.1_20_1.1.20.1+build.1-v2.jar:?]
	at net.minecraft.server.world.ServerWorld.method_31420(ServerWorld.java:356) ~[minecraft-merged-project-root-1.20.1-net.fabricmc.yarn.1_20_1.1.20.1+build.1-v2.jar:?]
	at net.minecraft.world.EntityList.forEach(EntityList.java:81) ~[minecraft-merged-project-root-1.20.1-net.fabricmc.yarn.1_20_1.1.20.1+build.1-v2.jar:?]
	at net.minecraft.server.world.ServerWorld.tick(ServerWorld.java:333) ~[minecraft-merged-project-root-1.20.1-net.fabricmc.yarn.1_20_1.1.20.1+build.1-v2.jar:?]
	... 6 more
[17:30:46] [Server thread/ERROR] (Minecraft) This crash report has been saved to: H:\minecraft modding\sculk_depths\run\crash-reports\crash-2023-07-16_17.30.46-server.txt
[17:30:46] [Server thread/INFO] (Minecraft) Stopping server
[17:30:46] [Server thread/INFO] (Minecraft) Saving players
---- Minecraft Crash Report ----
// Hey, that tickles! Hehehe!

Time: 2023-07-16 17:30:46
Description: Colliding entity with block

java.lang.NullPointerException: Cannot invoke "Object.hashCode()" because "key" is null
	at java.base/java.util.concurrent.ConcurrentHashMap.get(ConcurrentHashMap.java:936)
	at net.kyrptonaught.customportalapi.portal.linking.PortalLinkingStorage.getDestination(PortalLinkingStorage.java:50)
	at net.kyrptonaught.customportalapi.util.CustomTeleporter.customTPTarget(CustomTeleporter.java:65)
	at net.kyrptonaught.customportalapi.util.CustomTeleporter.TPToDim(CustomTeleporter.java:46)
	at net.kyrptonaught.customportalapi.CustomPortalBlock.onEntityCollision(CustomPortalBlock.java:110)
	at net.minecraft.block.AbstractBlock$AbstractBlockState.onEntityCollision(AbstractBlock.java:1106)
	at net.minecraft.entity.Entity.checkBlockCollision(Entity.java:1279)
	at net.minecraft.entity.Entity.tryCheckBlockCollision(Entity.java:1041)
	at net.minecraft.entity.Entity.move(Entity.java:997)
	at net.minecraft.entity.LivingEntity.applyMovementInput(LivingEntity.java:2156)
	at net.minecraft.entity.LivingEntity.travel(LivingEntity.java:2098)
	at net.minecraft.entity.LivingEntity.tickMovement(LivingEntity.java:2515)
	at net.minecraft.entity.mob.MobEntity.tickMovement(MobEntity.java:544)
	at net.minecraft.entity.passive.PassiveEntity.tickMovement(PassiveEntity.java:128)
	at net.minecraft.entity.passive.AnimalEntity.tickMovement(AnimalEntity.java:60)
	at net.minecraft.entity.passive.AbstractHorseEntity.tickMovement(AbstractHorseEntity.java:551)
	at net.minecraft.entity.LivingEntity.tick(LivingEntity.java:2249)
	at net.minecraft.entity.mob.MobEntity.tick(MobEntity.java:368)
	at net.minecraft.entity.passive.AbstractHorseEntity.tick(AbstractHorseEntity.java:580)
	at net.minecraft.server.world.ServerWorld.tickEntity(ServerWorld.java:654)
	at net.minecraft.world.World.tickEntity(World.java:481)
	at net.minecraft.server.world.ServerWorld.method_31420(ServerWorld.java:356)
	at net.minecraft.world.EntityList.forEach(EntityList.java:81)
	at net.minecraft.server.world.ServerWorld.tick(ServerWorld.java:333)
	at net.minecraft.server.MinecraftServer.tickWorlds(MinecraftServer.java:860)
	at net.minecraft.server.MinecraftServer.tick(MinecraftServer.java:798)
	at net.minecraft.server.integrated.IntegratedServer.tick(IntegratedServer.java:99)
	at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:658)
	at net.minecraft.server.MinecraftServer.method_29739(MinecraftServer.java:288)
Caused by: java.lang.NullPointerException: Cannot invoke "Object.hashCode()" because "key" is null

	at java.base/java.lang.Thread.run(Thread.java:833)


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Thread: Render thread
Stacktrace:
	at java.base/java.util.concurrent.ConcurrentHashMap.get(ConcurrentHashMap.java:936)
	at net.kyrptonaught.customportalapi.portal.linking.PortalLinkingStorage.getDestination(PortalLinkingStorage.java:50)
	at net.kyrptonaught.customportalapi.util.CustomTeleporter.customTPTarget(CustomTeleporter.java:65)
	at net.kyrptonaught.customportalapi.util.CustomTeleporter.TPToDim(CustomTeleporter.java:46)
	at net.kyrptonaught.customportalapi.CustomPortalBlock.onEntityCollision(CustomPortalBlock.java:110)
	at net.minecraft.block.AbstractBlock$AbstractBlockState.onEntityCollision(AbstractBlock.java:1106)
	at net.minecraft.entity.Entity.checkBlockCollision(Entity.java:1279)
	at net.minecraft.entity.Entity.tryCheckBlockCollision(Entity.java:1041)
	at net.minecraft.entity.Entity.move(Entity.java:997)
	at net.minecraft.entity.LivingEntity.applyMovementInput(LivingEntity.java:2156)
	at net.minecraft.entity.LivingEntity.travel(LivingEntity.java:2098)
	at net.minecraft.entity.LivingEntity.tickMovement(LivingEntity.java:2515)
	at net.minecraft.entity.mob.MobEntity.tickMovement(MobEntity.java:544)
	at net.minecraft.entity.passive.PassiveEntity.tickMovement(PassiveEntity.java:128)
	at net.minecraft.entity.passive.AnimalEntity.tickMovement(AnimalEntity.java:60)
	at net.minecraft.entity.passive.AbstractHorseEntity.tickMovement(AbstractHorseEntity.java:551)
	at net.minecraft.entity.LivingEntity.tick(LivingEntity.java:2249)
	at net.minecraft.entity.mob.MobEntity.tick(MobEntity.java:368)
	at net.minecraft.entity.passive.AbstractHorseEntity.tick(AbstractHorseEntity.java:580)
	at net.minecraft.server.world.ServerWorld.tickEntity(ServerWorld.java:654)
	at net.minecraft.world.World.tickEntity(World.java:481)
	at net.minecraft.server.world.ServerWorld.method_31420(ServerWorld.java:356)
	at net.minecraft.world.EntityList.forEach(EntityList.java:81)
	at net.minecraft.server.world.ServerWorld.tick(ServerWorld.java:333)

-- Block being collided with --
Details:
	Block: Block{customportalapi:customportalblock}[axis=z]
	Block location: World: (11,48,-12), Section: (at 11,0,4 in 0,3,-1; chunk contains blocks 0,-256,-16 to 15,127,-1), Region: (0,-1; contains chunks 0,-32 to 31,-1, blocks 0,-256,-512 to 511,127,-1)
Stacktrace:
	at net.minecraft.entity.Entity.checkBlockCollision(Entity.java:1279)
	at net.minecraft.entity.Entity.tryCheckBlockCollision(Entity.java:1041)
	at net.minecraft.entity.Entity.move(Entity.java:997)
	at net.minecraft.entity.LivingEntity.applyMovementInput(LivingEntity.java:2156)
	at net.minecraft.entity.LivingEntity.travel(LivingEntity.java:2098)
	at net.minecraft.entity.LivingEntity.tickMovement(LivingEntity.java:2515)
	at net.minecraft.entity.mob.MobEntity.tickMovement(MobEntity.java:544)
	at net.minecraft.entity.passive.PassiveEntity.tickMovement(PassiveEntity.java:128)
	at net.minecraft.entity.passive.AnimalEntity.tickMovement(AnimalEntity.java:60)
	at net.minecraft.entity.passive.AbstractHorseEntity.tickMovement(AbstractHorseEntity.java:551)
	at net.minecraft.entity.LivingEntity.tick(LivingEntity.java:2249)
	at net.minecraft.entity.mob.MobEntity.tick(MobEntity.java:368)
	at net.minecraft.entity.passive.AbstractHorseEntity.tick(AbstractHorseEntity.java:580)
	at net.minecraft.server.world.ServerWorld.tickEntity(ServerWorld.java:654)
	at net.minecraft.world.World.tickEntity(World.java:481)
	at net.minecraft.server.world.ServerWorld.method_31420(ServerWorld.java:356)
	at net.minecraft.world.EntityList.forEach(EntityList.java:81)
	at net.minecraft.server.world.ServerWorld.tick(ServerWorld.java:333)
	at net.minecraft.server.MinecraftServer.tickWorlds(MinecraftServer.java:860)
	at net.minecraft.server.MinecraftServer.tick(MinecraftServer.java:798)
	at net.minecraft.server.integrated.IntegratedServer.tick(IntegratedServer.java:99)
	at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:658)
	at net.minecraft.server.MinecraftServer.method_29739(MinecraftServer.java:288)
	at java.base/java.lang.Thread.run(Thread.java:833)

-- Entity being checked for collision --
Details:
	Entity Type: minecraft:mule (net.minecraft.entity.passive.MuleEntity)
	Entity ID: 1193
	Entity Name: Mule
	Entity's Exact location: 10.32, 48.00, -10.85
	Entity's Block location: World: (10,48,-11), Section: (at 10,0,5 in 0,3,-1; chunk contains blocks 0,-256,-16 to 15,127,-1), Region: (0,-1; contains chunks 0,-32 to 31,-1, blocks 0,-256,-512 to 511,127,-1)
	Entity's Momentum: 0.03, -0.00, -0.00
	Entity's Passengers: []
	Entity's Vehicle: null
Stacktrace:
	at net.minecraft.entity.Entity.tryCheckBlockCollision(Entity.java:1041)
	at net.minecraft.entity.Entity.move(Entity.java:997)
	at net.minecraft.entity.LivingEntity.applyMovementInput(LivingEntity.java:2156)
	at net.minecraft.entity.LivingEntity.travel(LivingEntity.java:2098)
	at net.minecraft.entity.LivingEntity.tickMovement(LivingEntity.java:2515)
	at net.minecraft.entity.mob.MobEntity.tickMovement(MobEntity.java:544)
	at net.minecraft.entity.passive.PassiveEntity.tickMovement(PassiveEntity.java:128)
	at net.minecraft.entity.passive.AnimalEntity.tickMovement(AnimalEntity.java:60)
	at net.minecraft.entity.passive.AbstractHorseEntity.tickMovement(AbstractHorseEntity.java:551)
	at net.minecraft.entity.LivingEntity.tick(LivingEntity.java:2249)
	at net.minecraft.entity.mob.MobEntity.tick(MobEntity.java:368)
	at net.minecraft.entity.passive.AbstractHorseEntity.tick(AbstractHorseEntity.java:580)
	at net.minecraft.server.world.ServerWorld.tickEntity(ServerWorld.java:654)
	at net.minecraft.world.World.tickEntity(World.java:481)
	at net.minecraft.server.world.ServerWorld.method_31420(ServerWorld.java:356)
	at net.minecraft.world.EntityList.forEach(EntityList.java:81)
	at net.minecraft.server.world.ServerWorld.tick(ServerWorld.java:333)
	at net.minecraft.server.MinecraftServer.tickWorlds(MinecraftServer.java:860)
	at net.minecraft.server.MinecraftServer.tick(MinecraftServer.java:798)
	at net.minecraft.server.integrated.IntegratedServer.tick(IntegratedServer.java:99)
	at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:658)
	at net.minecraft.server.MinecraftServer.method_29739(MinecraftServer.java:288)
	at java.base/java.lang.Thread.run(Thread.java:833)

-- Entity being ticked --
Details:
	Entity Type: minecraft:mule (net.minecraft.entity.passive.MuleEntity)
	Entity ID: 1193
	Entity Name: Mule
	Entity's Exact location: 10.32, 48.00, -10.85
	Entity's Block location: World: (10,48,-11), Section: (at 10,0,5 in 0,3,-1; chunk contains blocks 0,-256,-16 to 15,127,-1), Region: (0,-1; contains chunks 0,-32 to 31,-1, blocks 0,-256,-512 to 511,127,-1)
	Entity's Momentum: 0.03, -0.00, -0.00
	Entity's Passengers: []
	Entity's Vehicle: null
Stacktrace:
	at net.minecraft.world.World.tickEntity(World.java:481)
	at net.minecraft.server.world.ServerWorld.method_31420(ServerWorld.java:356)
	at net.minecraft.world.EntityList.forEach(EntityList.java:81)
	at net.minecraft.server.world.ServerWorld.tick(ServerWorld.java:333)
	at net.minecraft.server.MinecraftServer.tickWorlds(MinecraftServer.java:860)
	at net.minecraft.server.MinecraftServer.tick(MinecraftServer.java:798)
	at net.minecraft.server.integrated.IntegratedServer.tick(IntegratedServer.java:99)
	at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:658)
	at net.minecraft.server.MinecraftServer.method_29739(MinecraftServer.java:288)
	at java.base/java.lang.Thread.run(Thread.java:833)

-- Affected level --
Details:
	All players: 0 total; []
	Chunk stats: 961
	Level dimension: sculk_depths:sculk_depthsdim
	Derived: true
	Level spawn location: World: (0,111,0), Section: (at 0,15,0 in 0,6,0; chunk contains blocks 0,-256,0 to 15,127,15), Region: (0,0; contains chunks 0,0 to 31,31, blocks 0,-256,0 to 511,127,511)
	Level time: 1418405 game time, 1000 day time
	Level name: New World
	Level game mode: Game mode: creative (ID 1). Hardcore: false. Cheats: true
	Level weather: Rain time: 53722 (now: false), thunder time: 12108 (now: true)
	Known server brands: fabric
	Removed feature flags: 
	Level was modded: true
	Level storage version: 0x04ABD - Anvil
Stacktrace:
	at net.minecraft.server.MinecraftServer.tickWorlds(MinecraftServer.java:860)
	at net.minecraft.server.MinecraftServer.tick(MinecraftServer.java:798)
	at net.minecraft.server.integrated.IntegratedServer.tick(IntegratedServer.java:99)
	at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:658)
	at net.minecraft.server.MinecraftServer.method_29739(MinecraftServer.java:288)
	at java.base/java.lang.Thread.run(Thread.java:833)
```
```

**Comment by kyrptonaught** [comment]
This was fixed in `0.0.1-beta65-1.20.4`
